pkg/plugin/plugin.go

```diff
@@ -132,19 +132,23 @@ func RunPlugin(plugin Plugin) {
 	decoder := json.NewDecoder(os.Stdin)
 	var encoderMu sync.Mutex
 
+	wg := sync.WaitGroup{}
+
 	for {
 		var req rpc.Request
 		err := decoder.Decode(&req)
 		if err != nil {
 			if errors.Is(err, io.EOF) {
 				// plugin shutdown requested
-				return
+				break
 			}
 
 			log.Fatal("failed to read request:", err)
```

Comment:
If I'm not missing something, this would go to the stdout where the parent's `RPC` through the `resRead` pipe would read.

Thus, maybe configure logging to stderr or change logging to zap and let it log to stderr to match with the rest of the code base.

Reply:
The `log` package uses stderr by default: https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/log/log.go;l=87 (interestingly, this seems to be mentioned nowhere on https://pkg.go.dev/log)

You probably get some duplicate timestamps with that though.

Reply:
Oh thanks, my bad. I wrongly assumed that it would be logged to stdout.